### PR TITLE
chore(ec2): add missing instance types

### LINKED ISSUE #23854: aws_cdk.aws_ec2.InstanceClass: Support MAC2
### Describe the feature

I would like support for spinning up Mac ec2 Instances with ARM processors (M1).

### Use Case

Add it to the instance class definitions.

### Proposed Solution

Add `MAC2`(?) to the instance class definitions.

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

1.97.0 (looking to upgrade soon)

### Environment details (OS name and version, etc.)

Ubuntu 20.04

### PR DESCRIPTION
**Pull request by nmussy**
### Issue # (if applicable)

Closes #23854

### Reason for this change

Update the CDK EC2 instance class and size enums to match the current availability

### Description of changes

Added the missing instance class and sizes

### Description of how you validated changes

This was generated with the help of the same tool I used to create #29422, using the live SDK data to match against the CDK. For the instance types, I am using the `DescribeInstanceTypes` API command in multiple regions (`us-east-1`, `us-east-2`, `us-west-1`, `eu-west-1`.) As far as I could tell, the union of `us-east-1` and `us-east-2` cover all possible instance classes and sizes, but correct me if I'm wrong.

As a sanity check, I'm also checking that the CDK does not have more instance types and sizes that the SDK does. It does vary between regions, but of note, there were:
* `p4de`, in developer preview
* `x2g`, RDS exclusive and not returned by `DescribeInstanceTypes`

Finally, I am ignoring some of the previous generation instance classes missing from the CDK, as they are very outdated and weren't added in first place, i.e. `c1`, `t1`, `m1`, `m2`, and `i2`.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts [hunk 1]
```diff
@@ -1260,6 +1320,11 @@ export enum InstanceSize {
    */
   XLARGE56 = '56xlarge',
 
+  /**
+   * Instance size XLARGE56 (96xlarge)
```
Comment (msambol) [inline packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts:1324]: s/XLARGE56/XLARGE96

File: packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts [hunk 2]
```diff
@@ -1260,6 +1320,11 @@ export enum InstanceSize {
...
+   */
+  XLARGE96 = '96xlarge',
+
   /**
    * Instance size XLARGE56 (112xlarge)
```
Comment (msambol) [inline packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts:1329]: can you fix this while you're in here? `XLARGE56`

File: packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts [hunk 3]
```diff
@@ -1495,10 +1580,19 @@ export class InstanceType {
       [InstanceClass.INF2]: 'inf2',
       [InstanceClass.MACINTOSH1_INTEL]: 'mac1',
       [InstanceClass.MAC1]: 'mac1',
+      [InstanceClass.MACINTOSH2_M1]: 'mac2',
+      [InstanceClass.MAC2]: 'mac2',
+      [InstanceClass.MACINTOSH2_M2]: 'mac2-m2',
+      [InstanceClass.MAC2_M2]: 'mac2-m2',
+      [InstanceClass.MACINTOSH2_M2_PRO]: 'mac2-m2pro',
+      [InstanceClass.MAC2_M2PRO]: 'mac2-m2pro',
+
```
Comment (msambol) [inline packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts:1589]: extra newline

File: packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts [hunk 4]
```diff
@@ -1269,6 +1334,26 @@ export enum InstanceSize {
...
+  /**
+   * Instance size XLARGE32METAL (metal-32xl)
+   */
+  XLARGE32METAL = 'metal-32xl',
+
+  /**
+   * Instance size XLARGE48METAL (metal-48xl)
+   */
+  XLARGE48METAL = 'metal-48xl',
```
Comment (msambol) [inline packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts:1356]: do these not need to be in the list below?

**Review by msambol — CHANGES_REQUESTED** [review]
See inline comments.

File: packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts (hunk 1 above)
Comment (nmussy) [reply packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts:1324]: Good catch, fixed

File: packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts (hunk 2 above)
Comment (nmussy) [reply packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts:1329]: Done, thanks

File: packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts (hunk 3 above)
Comment (nmussy) [reply packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts]: Thanks, fixed

File: packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts (hunk 4 above)
Comment (nmussy) [reply packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts:1356]: If you mean in the `instanceClassMap`, they only contain instance classes, the `metal-*` properties are instance sizes

File: packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts (hunk 1 above)
Comment (msambol) [reply packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts:1324]: may have forgotten to push the change?

Comment (nmussy) [reply packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts]: Didn't stage them 🤦 Should be good now, sorry

File: packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts [hunk 5]
```diff
@@ -1130,6 +1160,36 @@ export enum InstanceClass {
...
+   */
+  HIGH_PERFORMANCE_COMPUTING6_INTEL_NVME_DRIVE = 'high-performance-computing6-intel-nvme-drive',
+
+  /**
+   * High performance computing with local NVME drive based on 6th generation with Intel Xeon Scalable processors (3rd generation processors code named Ice Lake), 6th generation
+   */
+  HPC6ID = 'hpc6id',
+
+  /**
+   * High performance computing based on Graviton, 7th generation
```
Comment (msambol) [inline packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts:1174]: this shouldn't be Graviton

File: packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts [hunk 6]
```diff
@@ -1130,6 +1160,36 @@ export enum InstanceClass {
...
+   */
+  HIGH_PERFORMANCE_COMPUTING7_AMD = 'high-performance-computing7-amd',
+
+  /**
+   * High performance computing based on Graviton, 7th generation
```
Comment (msambol) [inline packages/aws-cdk-lib/aws-ec2/lib/instance-types.ts:1179]: likewise here

**Review by msambol — APPROVED** [review]
LGTM 👍🏼

**Review by paulhcsun — APPROVED** [review]

**Pull request merged**